spec/unit/puppet/provider/powershell_spec.rb

```diff
@@ -0,0 +1,76 @@
+#! /usr/bin/env ruby
+require 'spec_helper'
+require 'puppet/type'
+require 'puppet/type/base_dsc'
+
+describe Puppet::Type.type(:base_dsc).provider(:powershell) do
+
+  # ROB - pick up here, the initialization is throwing on Windows if the symlink exists.
+  # The logic is likely slightly off
+
+  let (:resource) { Puppet::Type.type(:base_dsc).new(:provider => :powershell, :name => "windows_provider") }
+  let (:provider) { resource.provider }
+  let (:catalog) { Puppet::Resource::Catalog.new }
+
+  before :each do
+    resource.provider = provider
+  end
+
+  it "should be an instance of Puppet::Type::Base_dsc::ProviderPowershell" do
+    provider.must be_an_instance_of Puppet::Type::Base_dsc::ProviderPowershell
+  end
+
+  context ".set_symlink_folder" do
+    context "on Windows", :if => Puppet.features.microsoft_windows? do
+      before :each do
+        #delete the symlink if it exists
```

Comment:
This will be likely in the integration spec. Placeholder to remind myself to add.